.github/workflows/pip_install.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# A script to pip install a package with dependencies from requirements{suffix}.txt
+# if requirements{suffix}.txt exists.
+# Generates the requirements{suffix}.txt file once install is complete
+
+# usage: install_with_requirements.sh suffix [remaining parameters to pip install]
+
+suffix=$1
+shift
+
+touch requirements-${suffix}.txt
+pip install -r requirements-${suffix}.txt "${@}"
+mkdir -p lockfiles
+pip freeze --exclude-editable > lockfiles/requirements-${suffix}.txt
```

Comment:
Actually, the sed can be much shorter, should we duplicate it here and in the container step?
```suggestion
pip freeze --exclude-editable | sed '/file:/d' > lockfiles/requirements-${suffix}.txt
```

Reply:
Just deleting the 'file:' line from the requirements.txt will result in an empty file for python3-pip-skeleton. This causes the release to fail because you cant have empty assets. That's why we have the complicated sed and also repetition of sed is no longer an issue now I'm moving requirements.txt generation to the dist job.

Reply:
Resolved this by putting the substitution sed in the container job and having the simple deletion sed in the script which is used by lint and test jobs (both of which do dev installs)